integration_test/test_videoroom/test/integration/simulcast_test.exs

```diff
@@ -0,0 +1,323 @@
+      {:wait, 1_000},
+      {:click, @simulcast_inbound_stats, 10, 1_000, tag: :after_disabling_high_en},
+      {:wait, 1_000},
+      {:click, @simulcast_inbound_stats, 10, 1_000, tag: :after_disabling_low_en},
+      {:wait, 1_000},
+      {:click, @simulcast_inbound_stats, 10, 1_000, tag: :after_enabling_low_en},
+      {:wait, 1_000},
+      {:click, @simulcast_inbound_stats, 10, 1_000, tag: :after_enabling_high_en},
+      {:wait, 1_000},
+      {:click, @simulcast_outbound_stats, 10, 1_000, tag: :after_enabling_medium_en}
+    ]
+
+    actions_with_id = [actions1, actions2] |> Enum.with_index()
+
+    stage_to_expected_encoding = %{
+      after_disabling_medium_en: "h",
+      after_disabling_high_en: "l",
+      after_disabling_low_en: nil,
+      after_enabling_low_en: "l",
+      after_enabling_high_en: "h",
+      after_enabling_medium_en: "m"
+    }
+
+    for {actions, browser_id} <- actions_with_id, into: [] do
+      specific_mustang = %{
+        mustang_options
+        | id: browser_id,
+          actions: actions
+      }
+
+      Task.async(fn ->
+        Stampede.start({SimulcastMustang, specific_mustang}, @browser_options)
+      end)
+    end
+    |> Task.await_many(:infinity)
+
+    receive do
+      accumulated_stats ->
+        for stage <- Map.keys(stage_to_expected_encoding),
+            browser_id_to_iterated_stats = Map.get(accumulated_stats, stage) do
+          encoding = stage_to_expected_encoding[stage]
+
+          sender_iterated_stats = browser_id_to_iterated_stats[0]
+          receiver_iterated_stats = browser_id_to_iterated_stats[1]
+          stats = Enum.zip(sender_iterated_stats, receiver_iterated_stats)
+
+          case stage do
+            :after_disabling_low_en ->
+              assert(
+                Enum.any?(receiver_iterated_stats, &(&1["framesPerSecond"] == 0)),
+                "Failed on stage: #{stage} should be all encodings disabled,
+                receiver stats are #{inspect(receiver_iterated_stats)}
+                sender stats are #{inspect(sender_iterated_stats)}
+                "
+              )
+
+            _other ->
+              assert(
+                Enum.any?(stats, fn {all_encodings_sender_stats, receiver_stats} ->
+                  assert_stats_equal(receiver_stats, all_encodings_sender_stats[encoding]) and
+                    assert_receiver_encoding(receiver_stats, encoding)
+                end),
+                "Failed on stage: #{stage} should be encoding: #{encoding},
+                    receiver stats are: #{inspect(receiver_iterated_stats)}
+                    sender stats are: #{inspect(Enum.map(sender_iterated_stats, & &1[encoding]))}
+                    "
+              )
+          end
+        end
+    end
+  end
+
+  defp assert_receiver_encoding(receiver_stats, encoding) do
+    receiver_stats["encoding"] == encoding
+  end
+
+  defp assert_stats_equal(receiver_stats, sender_stats) do
```

Comment:
`sender_stats` -> `sender_encoding_stats` and let's write comment about receiver and sender point of view